modules/tensor_mechanics/include/materials/StressUpdateBase.h

```diff
@@ -91,6 +109,11 @@ class StressUpdateBase : public Material
 
   virtual Real computeTimeStepLimit();
 
+  virtual TangentCalculationMethod getTangentCalculationMethod()
```

Comment:
Once this gets merged and we fix up the models in the apps, I think I would like to not define this in the base class and make this pure virtual, but this is good for now.

Reply:
Are you sure? I think it's just fine to have all models start with providing nothing, and only (in the rare cases) that they calculate the tangent, then manually change it. It's safest if method is `ELASTIC` since it'll never dip into the `StressUpdateBase` models.

Reply:
Yeah, you're probably right.  Regardless, this is something for later anyway.

Reply:
👍